platform/integrations/karpenter.mdx

```diff
@@ -0,0 +1,449 @@
+          requirements:
+            - key: kubernetes.io/arch
+              operator: In
+              values: ["amd64"]
+            - key: kubernetes.io/os
+              operator: In
+              values: ["linux"]
+            - key: karpenter.sh/capacity-type
+              operator: In
+              values: ["spot"]
+            - key: karpenter.k8s.aws/instance-category
+              operator: In
+              values: ["c", "m", "r"]
+            - key: karpenter.k8s.aws/instance-generation
+              operator: Gt
+              values: ["2"]
+            - key: my.org/environ
+              operator: Exists
+          nodeClassRef:
+            apiVersion: karpenter.k8s.aws/v1beta1
+            kind: EC2NodeClass
+            name: <node-class-name>
+      limits:
+        cpu: 100
+      disruption:
+        consolidationPolicy: WhenUnderutilized
+        expireAfter: Never
+EOF
+    ```
+
+  </TabItem>
+  <TabItem value="AKS">
+
+    ```bash
+    cat <<EOF | envsubst | kubectl apply -f -
+    apiVersion: karpenter.sh/v1beta1
+    kind: NodePool
+    metadata:
+      name: default
+    spec:
+      template:
+        spec:
+          requirements:
+            - key: kubernetes.io/arch
+              operator: In
+              values: ["amd64"]
+            - key: kubernetes.io/os
+              operator: In
+              values: ["linux"]
+            - key: karpenter.sh/capacity-type
+              operator: In
+              values: ["on-demand"]
+            - key: karpenter.azure.com/sku-family
+              operator: In
+              values: [D]
+            - key: my.org/environ
+              operator: Exists
+          nodeClassRef:
+            name: <node-class-name>
+      limits:
+        cpu: 100
+      disruption:
+        consolidationPolicy: WhenUnderutilized
+        expireAfter: Never
+EOF
+    ```
+
+  </TabItem>
+</Tabs>
+
+2. Add nodeSelector to deployments in your virtual cluster using a nodeSelector that targets the label from step 1 with the desired value.
+
+```bash
+cat <<EOF | kubectl apply -f -
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: inflate
+spec:
+  replicas: 0
+  selector:
+    matchLabels:
+      app: inflate
+  template:
+    metadata:
+      labels:
+        app: inflate
+    spec:
+      nodeSelector:
+        my.org/environ: <any key, "prod", "dev", "test" make sense here>
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: inflate
+          image: public.ecr.aws/eks-distro/kubernetes/pause:3.7
+          resources:
+            requests:
+              cpu: 1
+EOF
+
+kubectl scale deployment inflate --replicas 5
+
+```
+
+If multiple deployments use the same key/value pair for an already deployed node they can be scheduled to it. Otherwise, a new node will be provisioned. You can read more about user-defined labels in the [Karpenter's User-Defined Labels docs](https://karpenter.sh/docs/concepts/scheduling/#user-defined-labels).
```

Comment:
> If multiple deployments use the same key/value pair for an already deployed node they can be scheduled to it.

I'm not sure what this is trying to say.

Reply:
If you make a deployment with key/value pair for an existing node it can be scheduled to that node. Otherwise, a new node will be provisioned.

Basically you're adding the key to the node pool, but using different values necessitate different nodes. I will word this differently.

Reply:
Added to separate commit. Mostly copied my explanations above.